src/env.cc

```diff
@@ -539,7 +539,8 @@ void Environment::AssignToContext(Local<v8::Context> context,
   context->SetAlignedPointerInEmbedderData(ContextEmbedderIndex::kRealm, realm);
   // Used to retrieve bindings
   context->SetAlignedPointerInEmbedderData(
-      ContextEmbedderIndex::kBindingListIndex, &(this->bindings_));
+      ContextEmbedderIndex::kBindingDataStoreIndex,
+      realm->binding_data_store());
```

Comment:
This function is called from `node_contextify.cc` like this:

```cc
env->AssignToContext(v8_context, nullptr, info);
```

This appears to get an invalid value in that flow. What's the expected behavior? Should the embedder data be `nullptr` when `realm` is `nullptr`?

Reply:
Afaict, the following might be the fix but I haven't fully looked into how this slot is used but I suspect the fix is:

```cc
  // Used to retrieve bindings
  context->SetAlignedPointerInEmbedderData(
      ContextEmbedderIndex::kBindingDataStoreIndex,
      realm != nullptr ? realm->binding_data_store() : nullptr);
```

Reply:
Sending https://github.com/nodejs/node/pull/48802 in case that is the fix.